src/coreComponents/mesh/mpiCommunications/NeighborCommunicator.cpp

```diff
@@ -620,5 +620,159 @@ void NeighborCommunicator::unpackBufferForSync( std::map< string, string_array >
   }
 }
 
+///////////////////////////  New versions of the packing and unpacking functions //////////////////////////////////////////////////////////////
+int NeighborCommunicator::packCommSizeForSync( std::map< fieldSyncKey, string_array > const & fieldsTobeSync,
+                                               MeshLevel const & mesh,
+                                               int const commID,
+                                               bool onDevice,
+                                               parallelDeviceEvents & events )
+{
+  GEOSX_MARK_FUNCTION;
+
+  NodeManager const & nodeManager = mesh.getNodeManager();
+  EdgeManager const & edgeManager = mesh.getEdgeManager();
+  FaceManager const & faceManager = mesh.getFaceManager();
```

Comment:
Maybe something like
```c++
std::map< Location, ObjectManagerBase const & > locToMgr{
  { Location::Node, mesh.getNodeManager() },
  { Location::Edge, mesh.getEdgeManager() },
  { Location::Face, mesh.getFaceManager() }
}
```
(or maybe with a `ObjectManager const *` or with `auto locToMgr{...}`) followed by
```c++
    switch( key.location )
    {
      case Location::Node:
      case Location::Edge:
      case Location::Face:
        bufferSize += locToMgr.at( key.location ).packSize( fields.second, nodeGhostsToSend, 0, onDevice, events );
        break;
      case Location::Elem:
        ...
```
?

Reply:
BTW, it's a bit strange to extract some info `mgr->getNeighborData( m_neighborRank ).ghostsToSend()` from the `mgr` in order to give it back. Maybe we can do something better, but maybe out of scope too.

Reply:
We could probably only pass the `neighborRank` to the the `mgr`. This change would be quite invasive though ( I think)